src/shared_modules/indexer_connector/src/indexerConnector.cpp

```diff
@@ -12,50 +12,74 @@
 #include "indexerConnector.hpp"
 #include "HTTPRequest.hpp"
 #include "shared_modules/indexer_connector/src/serverSelector.hpp"
+#include <fstream>
 
 // TODO: remove the LCOV flags when the implementation of this class is completed
 // LCOV_EXCL_START
 std::unordered_map<IndexerConnector*, std::unique_ptr<ThreadDispatchQueue>> QUEUE_MAP;
 constexpr auto DATABASE_WORKERS = 1;
+constexpr auto DATABASE_BASE_PATH = "queue/indexer/";
 
-IndexerConnector::IndexerConnector(const nlohmann::json& config)
+IndexerConnector::IndexerConnector(const nlohmann::json& config, const std::string& templatePath)
 {
     // Initialize publisher.
-    auto selector = std::make_shared<ServerSelector>(config.at("servers"));
+    auto selector = std::make_shared<ServerSelector>(config.at("indexer").at("hosts"));
```

Comment:
This allocation memory could be done after checking if the template file has been opened correctly.